docs/style-guide.md

```diff
@@ -0,0 +1,1348 @@
+[TOC]
+
+About this guide
+================
+
+This guide is mostly based on the
+[Google C++ Style Guide](https://google.github.io/styleguide/cppguide.html).
+
+> Sometimes the Google rules are tweaked, sometimes contradicted.
+>
+> This guide includes rules beyond those in the Google guide.
+
+Other notable sources of best software engineering practices include:
+
+* [JSF Air Vehicle C++ Coding Standards](http://www.stroustrup.com/JSF-AV-rules.pdf),
+* [Code Complete](http://cc2e.com) by Steve McConnell,
+* [Source codes of the Trilinos project](https://trilinos.org),
+* [Microsoft .Net Guidelines](https://msdn.microsoft.com/en-us/library/ms229042(v=vs.110).aspx),
+* [Linux Kernel Coding Style](https://www.kernel.org/doc/Documentation/CodingStyle).
+
+Some conventions introduced in this guide originate from:
+
+* NVIDIA CUDA,
+* Intel MKL,
+* Microsoft.
+
+This guide is created using Markdown.
+For Markdown documentation consult:
+
+* http://daringfireball.net/projects/markdown/,
+* https://en.wikipedia.org/wiki/Markdown,
+
+General Guidelines
+==================
+
+* Be consistent in your own code.
+* Follow the conventions already established by the project.
+* If you spot inconsistencies, fix them.
+* Break rules if it helps readability. This is only a guide.
+
+> This guide was written for PLASMA. Other ICL projects have their own
+> conventions. A few differences with SLATE's style are noted here.
+
+Standard Compliance
+===================
+
+C codes should be C99 compliant and compiled with the `-std=c99` flag,
```

Comment:
I'd like to revisit this in the future. C17 is safe now and many C23 features already made it to GCC and LLVM (and others that now rely on LLVM front end). Windows lack compliant C compiler anyway (try complex floating point, for example) so it's not a concern for now. But I digress...